### PR TITLE
adding connection_status

### PR DESCRIPTION
**Pull request by philbucher**
now the `connection_status` should be working as you describe in the tutorial

### CODE REVIEW

File: co_sim_io/impl/connection.hpp [hunk 1]
```diff
@@ -55,14 +55,18 @@ class Connection
 
     ReturnInfo Connect()
     {
-        mpComm->Connect();
-        return ReturnInfo(); // TODO use this, should put a return code based on the bool that comes back from mpComm->Connect()
+        const bool is_connected = mpComm->Connect();
+        ReturnInfo ret_info;
+        ret_info.Set<bool>("connection_status", is_connected);
```
Comment (pooyan-dadvand) [inline co_sim_io/impl/connection.hpp:60]: I thought that you had an int with different options.

Comment (philbucher) [reply co_sim_io/impl/connection.hpp]: sure, changed it :man_facepalming:

Comment (pooyan-dadvand) [reply co_sim_io/impl/connection.hpp]: I would also add some constexpr status values like:
- NotConnected: Connect is not called yet
- Connected: after successful call to Connect
- Disconnected: after successful call to Disconnect
- ConnectionError : Having a error message in return_info
- DisconnectionError: Having an error message in return_info

Comment (philbucher) [reply co_sim_io/impl/connection.hpp]: How would this work for C?

Comment (pooyan-dadvand) [reply co_sim_io/impl/connection.hpp]: enums, or const int values

Comment (philbucher) [reply co_sim_io/impl/connection.hpp]: Wouldn't it then make more sense to use enums also in C++?

Comment (pooyan-dadvand) [reply co_sim_io/impl/connection.hpp]: As you like, there is an implicit conversion from enum to int in both languages

**Pull request merged**